### DISCUSSION TITLE
After Linux reinstall getting Bad Gateway

### DISCUSSION THREAD

**Discussion by r-dannhauer** [pretix/pretix#2248] [Q&A] [answered]
**Describe the bug**
Pretix Control (or Pretix in general) generates a 502 Bad Gateway after a fresh install with an already filled MySQL database and importing the "old" Config. Same Server, just after an Ubuntu Upgrade.

**Expected behavior**
Not do that.

**Screenshots**
We all know how a Bad Gateway from nginx looks like.

**Additional context**
I added the needed nginx entries from the old config (modified, of course). I get following error
2021/10/12 01:03:16 [error] 214573#0: *1 no live upstreams while connecting to upstream, client: [myip], server: [mydomain], request: "GET /control/ HTTP/1.1", upstream: "http://localhost/control/", host: "[mydomain]"
Checking the open ports Redis doesn't seem to have that port binded. Restarting nginx, redis and the pretix workers and web services did not help.

**Comment by raphaelm** [comment, marked as answer]
If redis is not running, that's obiously one thing to dig deeper into.

Also, please have a look into the log output of the pretix-web.service as well as pretix' log files if there are obvious errors why pretix is not running.

(Converting this to a discussion since it's almost certainly not an issue with pretix but somewhere in the setup.)

**Reply by r-dannhauer** [reply]
Well, Redis is running (checking with service redis-server status).
Do you mean those under /var/pretix/data/logs? Both are empty. As in created, no content (csp.log and pretix.log)

I'm actually thinking about getting Pretix in a Docker container, though I have 0 experience in Docker.
//EDIT: Ya, turns out I need Redis also if I want to use Pretix via Docker.

**Reply by r-dannhauer** [reply]
Okay, found the logs you meant, sorry.
Web Service is exited due to a failure:
Okt 12 13:08:58 [host] gunicorn[258026]:   File "/var/pretix/venv/lib/python3.8/site-packages/gunicorn/arbiter.py", line 342, in halt
Okt 12 13:08:58 [host] gunicorn[258026]:     self.stop()
Okt 12 13:08:58 [host] gunicorn[258026]:   File "/var/pretix/venv/lib/python3.8/site-packages/gunicorn/arbiter.py", line 393, in stop
Okt 12 13:08:58 [host] gunicorn[258026]:     time.sleep(0.1)
Okt 12 13:08:58 [host] gunicorn[258026]:   File "/var/pretix/venv/lib/python3.8/site-packages/gunicorn/arbiter.py", line 242, in handle_chld
Okt 12 13:08:58 [host] gunicorn[258026]:     self.reap_workers()
Okt 12 13:08:58 [host] gunicorn[258026]:   File "/var/pretix/venv/lib/python3.8/site-packages/gunicorn/arbiter.py", line 525, in reap_workers
Okt 12 13:08:58 [host] gunicorn[258026]:     raise HaltServer(reason, self.WORKER_BOOT_ERROR)
Okt 12 13:08:58 [host] gunicorn[258026]: gunicorn.errors.HaltServer: <HaltServer 'Worker failed to boot.' 3>
Okt 12 13:08:58 [host] systemd[1]: pretix-web.service: Main process exited, code=exited, status=1/FAILURE

Pretix-worker seems to be running fine

**Reply by raphaelm** [reply]
Can you show more of this log? Maybe it shows a reason for the exit

**Reply by r-dannhauer** [reply]
As I stated in another post on this discussion I could solve it by giving pretix chown rights on /var/pretix. Now I have the problem that anything under /media doesn't load, as the media doesn't get generated. My browser tries to load the css of the event, but all I see is the css of the event holder, no sub folder. I fear that my right management is wrong. What do I need to set the rights to?

**Comment by r-dannhauer** [comment]
Okay, fixed it. Solution was to set chown of /var/pretix to pretix:pretix. Fixed my issue. Now I'm fighting with stylesheets as for some reasons their filenames have changed.

**Reply by raphaelm** [reply]
Did you copy the contents of /media from your old system?

**Reply by r-dannhauer** [reply]
No, I did not.
//EDIT: i should not write in two threads in a discussion at once. They don't get generated, probably after I changed the chown.

**Reply by raphaelm** [reply]
Hmm, do you have a backup? There's no official way of recovering /media, it's not generally something that can be "generated" and should be backed up:

https://docs.pretix.eu/en/latest/admin/maintainance.html#backups

The CSS files are the exception, `python -m pretix updatestyles` might re-generate them, but also might not do so until the next pretix update because the caching mechanism thinks the CSS files are already up to date.

But you will be missing all other kinds of stuff, such as event logos, product pictures, ticket design backgrounds, invoice PDF files, etc.

**Reply by r-dannhauer** [reply]
Oops, good to know. That explains why in a testrun on a different system it worked, because I did an update of Pretix at the same time. But this time it failed, probably because of not having set up everything correctly.
But why did it generate the css file of the event holder then?
`python -m pretix updatestyles` got my machine working for 30 seconds or so, but nothing new under /media. I still believe I have the wrong chown settings there.

**Reply by r-dannhauer** [reply]
Restored my old media file (had to do a full recovery, Strato makes daily backups thankfully), everything's working fine now. Thank you! :)
